src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3078,6 +3086,72 @@ private BoundExpression BindArgumentExpression(BindingDiagnosticBag diagnostics,
                         arguments[arg] = BindToNaturalType(argument, diagnostics);
                     }
                 }
+
+                uint getOrCalculateInferredOutSafeToEscape()
+                {
+                    if (lazyInferredOutArgumentSafeToEscape is { } inferredOutArgumentSafeToEscape)
+                    {
+                        return inferredOutArgumentSafeToEscape;
+                    }
+
+                    // the safe-to-escape of any out parameters in the method we are calling.
+                    uint outParamSafeToEscape = UseUpdatedEscapeRulesForInvocation(leastOverriddenMember) ? ReturnOnlyScope : CallingMethodScope;
+
+                    // the narrowest safe-to-escape of any argument which can escape (either by-value or by-reference) into the argument for any out parameter.
+                    uint narrowestArgSafeToEscape = CallingMethodScope;
+
+                    // TODO: try and reuse arg-mixing bits to reduce redundant logic here
```

Comment:
sharing the "get escape values" logic is tricky here because several collections which ValueChecks wants as immutable are still builders in here. Overload resolution is done here, but we haven't bound default arguments (which I don't think matters for escape, since constants can always escape anywhere).

Maybe we should move this inference to occur *during* argument mixing checks somehow?

Reply:
I think effectively re-implsementing teh mixing logic here is fragile, I'd prefer we just refactored the underlying code to accept both types of inputs.

Reply:
Why would we not just add the inference to the start of `CheckInvocationArgMixing`? That is the choke point of our analysis and it's called on all method invocation paths. The only downside I could see is that we end up possibly doing inference twice but that should be fine cause it's idempotent